### PR TITLE
Check CSRF token in header too

### PR DESCRIPTION
**Pull request by olleharstedt**
Forum thread: https://forum.yiiframework.com/t/checking-cookie-for-csrf-token/132655

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️/❌
| Fixed issues  | -

### CODE REVIEW

File: framework/web/CHttpRequest.php [hunk 1]
```diff
@@ -1364,6 +1365,12 @@ public function validateCsrfToken($event)
 					$maskedUserToken=$this->getDelete($this->csrfTokenName);
 			}
 
+            // @see https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#jquery
+            // Assumes header is named like Yii-Csrf-Token if csrfTokenName is YII_CSRF_TOKEN
+            if (empty($maskedUserToken)) {
+                $maskedUserToken = $_SERVER['HTTP_' . $this->csrfTokenName];
```
Comment (samdark) [inline framework/web/CHttpRequest.php:1371]: Woudn't it error here if there's no header?

Comment (olleharstedt) [reply framework/web/CHttpRequest.php:1371]: Yep, will fix.

**Review by marcovtwout — CHANGES_REQUESTED** [review]
Current PR does not adhere to yii code style (indentation, brackets)

**Comment by marcovtwout** [comment]
I'm not sure yet this qualifies as a good generic framework addition. It relies on a specific header name but this feature is not used by yii framework code. Alternatively users can simply extend CHttpRequest->validateCsrfToken() to add the header check.

**Comment by samdark** [comment]
It's not specific header name, it is configurable and matches CSRF token name.

**Comment by marcovtwout** [comment]
(Note parallel implementation in Yii2 https://www.yiiframework.com/doc/api/2.0/yii-web-request#getCsrfTokenFromHeader()-detail)

**Pull request closed without merging**